### ISSUE TITLE
aria-invalid={true/false} incorrectly fails check

### ISSUE THREAD

**Issue by fracmak** [evcohen/eslint-plugin-jsx-a11y#125]
When specifying aria-invalid="false" or aria-invalid="true", the value gets converted into a boolean by getLiteralPropValue(), which fails the token check because it's looking for a "string".

**Comment by ljharb** [comment]
`aria-invalid={true}` is not a valid value - that will end up as `aria-invalid="aria-invalid"`, which is not a correct spec value. You must provide the literal *string* - not boolean - `"true"` or `"false"`.

If I'm misunderstanding your issue, please provide some code so I can unconfuse myself :-)

**Comment by fracmak** [comment]
That's what I did and it still failed. `aria-invalid="true"` as far as I can tell the getLiteralPropValue() function is recognizing the "true" as a Boolean and casting it out of the string. No way I can find around it.

**Comment by ljharb** [comment]
aha, thanks for clarifying.

**Comment by evcohen** [comment]
ah good find, thanks! basically, the ARIA properties needs to be able to handle multiple types. In this case, aria-invalid can be either a token or a boolean. would you be interested in trying to fix this bug? i'd be happy to help you along the way!

**Comment by fracmak** [comment]
Sure, what did you have in mind to fix this? One option I had considered was removing the constraint that all token values had to be string, and instead matching that the token value merely === the value stored in ARIA.json, I'd then update ARIA-INVALID's value's to be `[ "grammar", "false", "spelling", "true", false, true ]`

**Comment by evcohen** [comment]
@fracmak that seems like a good fix, although i don't think the string values "true" and "false" will ever be hit. `[ "grammar", "spelling", true, false ]` should be sufficient. 

the equality check is [here](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/src/rules/aria-proptypes.js#L45)

**Comment by fracmak** [comment]
Fixed in #126

**Comment by evcohen** [comment]
published in [v3.0.2](https://github.com/evcohen/eslint-plugin-jsx-a11y/releases/tag/3.0.2)! thanks for reporting and your help fixing!

**Comment by adillokili** [comment]
hello pleas of some one have a souloution for aria invalide = true i need to convert it  to false thank you
